### ISSUE TITLE
[FR] Multi-label dataset stratification

### ISSUE THREAD

**Issue by ehofesmann** [voxel51/fiftyone#897]
A natural use case of FiftyOne is to take a labeled dataset and create evenly stratified train/test/validation splits.

This is easy to do for single label classification datasets, however, for multi-label classifications or detections, this problem is much more difficult.

A function like `dataset.stratify(splits=["train","test","validation"], ratios=[0.7, 0.2, 0.1])` similar to [what is done in sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedShuffleSplit.html) but that also works for detections would be useful.
